scripts/multinode.sh

```
@@ -0,0 +1,153 @@
+
+  # Clear a temp environment file in-case we're already on the deployment node
+  > /tmp/env
+  echo "export ANSIBLE_FORCE_COLOR=${ANSIBLE_FORCE_COLOR}" >> script_env
+  scp script_env $infra01:/tmp/env
+
+  ssh root@$infra01 "source /tmp/env; ${command}"
+}
+
+function run_tag {
+  export ANSIBLE_FORCE_COLOR
+  local tag="$1"
+
+  echo "Running ${tag} from multinode.yml"
+
+  ansible-playbook \
+    --inventory-file="inventory/${LAB_PREFIX}-${LAB}" \
+    --extra-vars="@vars/${LAB_PREFIX}-${LAB}.yml" \
+    --extra-vars="product_repo_dir=${PRODUCT_REPO_DIR}" \
+    --extra-vars="osad_repo_dir=${OSAD_REPO_DIR}" \
+    --extra-vars="product_url=${PRODUCT_URL}" \
+    --extra-vars="product_branch=${PRODUCT_BRANCH}" \
+    --extra-vars="config_prefix=${CONFIG_PREFIX}" \
+    --tags="${PRODUCT},${tag},${LAB_PREFIX}" \
+    ${ANSIBLE_OPTIONS} \
+    ${LAB_PREFIX}-multinode.yml
+}
+
+function run_script {
+  local script="$1"
+
+  echo "Running ${script} from ${REPO_DIR}/scripts."
+  ssh_command "cd ${REPO_DIR}; bash scripts/${script}.sh"
+}
+
+function prepare {
+  echo "Sleep 3 minutes for reboot"
+  sleep 180
+
+  run_tag prepare
+}
+
+function run {
+  echo "Sleep 3 minutes for reboot"
+  sleep 180
+
+  echo "export DEPLOY_MAAS=${DEPLOY_MAAS}" > script_env
+  echo "export DEPLOY_HAPROXY=${DEPLOY_HAPROXY}" > script_env
+  run_script $BUILD_SCRIPT_NAME
+}
+
+function upgrade {
+  run_tag upgrade
+  run_script $UPGRADE_SCRIPT_NAME
+}
+
+function test {
+  echo "export TEMPEST_SCRIPT_PARAMETERS=${TEMPEST_SCRIPT_PARAMETERS}" > script_env
+
+  # Due to rpc-openstack having osad as a git submodule
+  # when we run the tests for deploying from rpc-openstack
+  # we have to change the DIR that we start from
+  REPO_DIR=$OSAD_REPO_DIR
+  run_script $TEST_SCRIPT_NAME
+}
+
+function rekick {
+  run_tag rekick
+}
+
+function cleanup {
+  run_tag cleanup
+}
+
+function main {
+  local retval
+
+  # Variables based products' respective documentation
```

What do you mean by different types, exactly? I changed the product names to be more specific. Also, I included an `elif` because I thought it should fail hard if the name was improperly set rather than fail later on down the road.